pygradle-plugin/src/test/groovy/com/linkedin/gradle/python/tasks/action/PipActionHelpers.groovy

```diff
@@ -0,0 +1,116 @@
+
+import com.linkedin.gradle.python.util.DefaultPackageSettings
+import com.linkedin.gradle.python.util.PackageInfo
+import org.junit.rules.TemporaryFolder
+
+import java.nio.file.Paths
+
+class PipActionHelpers {
+    private PipActionHelpers() {
+        //NOOP
+    }
+
+    static class RequiresRebuildOverridePackageSettings extends DefaultPackageSettings {
+        private final List<String> override
+
+        RequiresRebuildOverridePackageSettings(TemporaryFolder temporaryFolder, List<String> override) {
+            super(temporaryFolder.root)
+            this.override = override
+        }
+
+        @Override
+        boolean requiresSourceBuild(PackageInfo packageInfo) {
+            return override.contains(packageInfo.name)
+        }
+    }
+
+    static class SupportedLanguageVersionOverridePackageSettings extends DefaultPackageSettings {
+        private final Map<String, List<String>> override
+
+        SupportedLanguageVersionOverridePackageSettings(TemporaryFolder temporaryFolder, Map<String, List<String>> override) {
+            super(temporaryFolder.root)
+            this.override = override
+        }
+
+        @Override
+        List<String> getSupportedLanguageVersions(PackageInfo packageInfo) {
+            return override.getOrDefault(packageInfo.name, [])
+        }
+    }
+
+    static class InstallOptionOverridePackageSettings extends DefaultPackageSettings {
+        private final Map<String, List<String>> override
+
+        InstallOptionOverridePackageSettings(TemporaryFolder temporaryFolder, Map<String, List<String>> override) {
+            super(temporaryFolder.root)
+            this.override = override
+        }
+
+        @Override
+        List<String> getInstallOptions(PackageInfo packageInfo) {
+            return override.getOrDefault(packageInfo.name, [])
+        }
+    }
+
+    static class GlobalOptionOverridePackageSettings extends DefaultPackageSettings {
+        private final Map<String, List<String>> override
+
+        GlobalOptionOverridePackageSettings(TemporaryFolder temporaryFolder, override) {
+            super(temporaryFolder.root)
+            this.override = override
+        }
+
+        @Override
+        List<String> getGlobalOptions(PackageInfo packageInfo) {
+            return override.getOrDefault(packageInfo.name, [])
+        }
+    }
+
+    static class EnvOverridePackageSettings extends DefaultPackageSettings {
+        private final Map<String, Map<String, String>> envOverride
+
+        EnvOverridePackageSettings(TemporaryFolder temporaryFolder, Map<String, Map<String, String>> envOverride) {
+            super(temporaryFolder.root)
+            this.envOverride = envOverride
+        }
+
+        @Override
+        Map<String, String> getEnvironment(PackageInfo packageInfo) {
+            envOverride.getOrDefault(packageInfo.name, [:])
+        }
+    }
+
+    static class BuildOptionsPackageSettings extends DefaultPackageSettings {
```

Comment:
For consistency, this should also be a singular "Option" (or other names need to have plural), and should have Override in the name as other classes do.